src/hackrf/__tests__/README.md

```diff
@@ -0,0 +1,246 @@
+- **`HackRFErrorRecovery.test.ts`** - Error handling and recovery logic tests
+- **`HackRFMocked.test.ts`** - Additional mocked tests for edge cases and protocol compliance
+
+These tests run in CI/CD and don't require physical hardware. They use Jest mocks to simulate USB device behavior and verify:
+
+- USB control transfer formatting
+- Configuration validation
+- State management
+- Error handling
+- Buffer management
+- Protocol compliance
+
+### 2. Hardware-Gated Tests (Conditional)
+
+- **`HackRFHardware.test.ts`** - Tests that require a physical HackRF device
+
+These tests only run when:
+
+1. `HACKRF_HARDWARE_TESTS=true` environment variable is set
+2. HackRF command-line tools are installed (`hackrf_info`, `hackrf_transfer`)
+3. A physical HackRF device is connected
+
+Hardware tests verify:
+
+- Device detection using native commands
+- Data streaming using `hackrf_transfer`
+- Device information retrieval
+- Cross-spectrum frequency testing
+- Sample count handling
+
+## Running Tests
+
+### Run All Tests (Mocked Only)
+
+```bash
+npm test -- src/hackrf/__tests__/
+```
+
+### Run Specific Test Suite
+
+```bash
+# Unit tests with mocks
+npm test -- src/hackrf/__tests__/HackRFOne.test.ts
+npm test -- src/hackrf/__tests__/HackRFOneAdapter.test.ts
+npm test -- src/hackrf/__tests__/HackRFMocked.test.ts
+
+# Hardware tests (requires device)
+HACKRF_HARDWARE_TESTS=true npm test -- src/hackrf/__tests__/HackRFHardware.test.ts
+```
+
+### Run with Coverage
+
+```bash
+npm test -- src/hackrf/__tests__/ --coverage
+```
+
+## Hardware Test Setup
+
+### Prerequisites
+
+1. **Install HackRF Tools**
+
+   ```bash
+   # Ubuntu/Debian
+   sudo apt-get install hackrf
+
+   # macOS
+   brew install hackrf
+
+   # Arch Linux
+   sudo pacman -S hackrf
+   ```
+
+2. **Connect HackRF Device**
+
+   Connect your HackRF One via USB.
+
+3. **Verify Device Detection**
+
+   ```bash
+   hackrf_info
+   ```
+
+   Expected output:
+
+   ```
+   Found HackRF
+   Board ID Number: 2 (HackRF One)
+   Firmware Version: 2021.03.1 (API:1.04)
+   Part ID Number: 0xa000cb3c 0x00524f4b
+   Serial Number: 0x0000000000000000088869dc2b7c125f
+   ```
+
+4. **Test Data Streaming**
+
+   ```bash
+   hackrf_transfer -r /dev/null -f 100000000 -n 1000000
+   ```
+
+### Running Hardware Tests
+
+Enable hardware tests with the environment variable:
+
+```bash
+HACKRF_HARDWARE_TESTS=true npm test -- src/hackrf/__tests__/HackRFHardware.test.ts
+```
+
+### CI/CD Integration
+
+In CI/CD environments without physical hardware, hardware tests automatically skip. The test suite includes informational tests that always run to report the test environment status:
+
+```
+=== HackRF Hardware Test Environment ===
+HACKRF_HARDWARE_TESTS: false (not set)
+hackrf_info available: no
+Device connected: no
+
+To enable hardware tests:
+  HACKRF_HARDWARE_TESTS=true npm test
+=========================================
+```
+
+## Test Coverage
+
+Current coverage for HackRF implementation:
```

Comment:
Do not add documentation with derived data. It will change quickly and become outdated